staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -195,6 +195,16 @@ func (g *genValidations) hasValidationsMiss(n *typeNode, seen map[*typeNode]bool
 		if g.hasValidationsImpl(c.node, seen) {
 			return true
 		}
+		for _, inner := range c.inner {
+			if !inner.fieldValidations.Empty() {
+				return true
+			}
+		}
+		for _, inner := range c.inner {
+			if g.hasValidationsImpl(inner.node, seen) {
```

Comment:
Do we really need to recurse here?  These child nodes are specially constructed to represent "inner" which is exactly one level deep.  Am I misunderstanding?  either way, this warrants a comment ;)

Reply:
Good point, I initially based this off the regular node case above in this method but the recursion is not necessary for inner as noted.  I've removed it now and added a comment:

```
			// c.inner is created during parsing when a +k8s:inner tag is found so we
			// can assume there is a validation and don't need to recurse on the node
```